rexrov2_description/launch/upload.launch.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+import launch
+import launch_ros.actions
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+from launch.actions import ExecuteProcess, IncludeLaunchDescription
+from launch import LaunchDescription
+from launch.substitutions import LaunchConfiguration, ThisLaunchFileDir
+from ament_index_python.packages import get_package_share_directory
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+
+import xacro
+
+
+def generate_launch_description():
+    
+    x = LaunchConfiguration('x', default='0.0')
+    y = LaunchConfiguration('y', default='0.0')
+    z = LaunchConfiguration('z', default='1.0')
+    pkg_share = FindPackageShare('rexrov2_description').find('rexrov2_description')
+    robots_dir = os.path.join(pkg_share, 'robots')
+    xacro_file = os.path.join(robots_dir, 'rexrov2_default.xacro')
```

Comment:
I think it is better to avoid resolving the substitutions here. So, I think it is better not to call `.find(...)` and prefer `PathJoinSubstitution` instead of `os.path.join`. Something like:
```python
xacro_file = PathJoinSubstitution([FindPackageShare('rexrov2_description'), 'robots/rexrov2_default.xacro'])
```

Reply:
I tried but I think that, if we use `PathJointSubstitution` the xacro process gives an error

Reply:
The idea is to call the `xacro` executable with the Command action like [these](https://github.com/orise-robotics/diffbot2_simulation/blob/c7405faa9d3b5f8a124079e2fdeebffcccdf91b0/launch/diffbot2.launch.py#L102) [examples](https://github.com/orise-robotics/xacro_live/blob/f63adbdb190160ca3a2aa982b8913192bc8b2394/launch/xacro_live_view.launch.py#L65-L68). It would be something like:
```python
Command([ExecutableInPackage(package='xacro', executable='xacro'), xacro_file])
```

Reply:
@mateus-amarante, like @pxalcantara said, this approach using `PathJoinSubstitution` and `Command([ExecutableInPackage(package='xacro', executable='xacro'), xacro_file])`
 is not working. For now, I'm using the `xacro.process(urdf_file)` as suggested by @pxalcantara .